modules/ingestor/src/graph/product/mod.rs

```diff
@@ -35,10 +37,16 @@ impl<'g> ProductContext<'g> {
             version: Set(version.clone()),
         };
 
-        Ok(ProductVersionContext::new(
-            self,
-            model.insert(&self.graph.connection(&tx)).await?,
-        ))
+        //TODO: what about duplicate versions?
+        let product_version =
```

Comment:
So here's where we'd have graph.ingest_product(...) that would either fetch an existing (and wrap it in a Context) or insert a new (and wrap it in a Context).  Stick the de-dupe logic into graph::ingest_product().

"ingest" is basically "upsert" in graph-speak, confusingly.

Reply:
Added some logic to check for the existing version and update the sbom if needed. I hoped that I could have done it only on conflict, but it seems sea orm is not yet there.